picoquic/sender.c

```diff
@@ -2142,14 +2142,17 @@ int picoquic_prepare_packet(picoquic_cnx_t* cnx,
             break;
         }
         else {
-            ret = picoquic_prepare_segment(cnx, path_x, packet, current_time, 
+            ret = picoquic_prepare_segment(cnx, path_x, packet, current_time,
                 send_buffer + *send_length, available, &segment_length);
 
             if (ret == 0) {
                 *send_length += segment_length;
                 if (packet->length == 0 ||
                     packet->ptype == picoquic_packet_1rtt_protected_phi0 ||
                     packet->ptype == picoquic_packet_1rtt_protected_phi1) {
+                    if (packet->length == 0) {
+                        free(packet);
+                    }
                     break;
```

Comment:
Weird. I am pretty sure I coded exactly that in my recent patch. But you are correct, it did not make it into the PR. Something went wrong. Thanks for fixing it.